src/kaleidoscope/plugin/OneShot.cpp

```diff
@@ -22,266 +22,339 @@
 namespace kaleidoscope {
 namespace plugin {
 
-// ---- state ---------
+// ----------------------------------------------------------------------------
+// Configuration variables
 
-uint16_t OneShot::start_time_ = 0;
 uint16_t OneShot::time_out = 2500;
 uint16_t OneShot::hold_time_out = 250;
 int16_t OneShot::double_tap_time_out = -1;
-OneShot::key_state_t OneShot::state_[OneShot::ONESHOT_KEY_COUNT];
-Key OneShot::prev_key_;
-bool OneShot::should_cancel_ = false;
-bool OneShot::should_cancel_stickies_ = false;
 
-bool OneShot::isPressed() {
-  for (uint8_t i = 0; i < ONESHOT_KEY_COUNT; i++) {
-    if (state_[i].pressed)
+// ----------------------------------------------------------------------------
+// State variables
+
+OneShot::OneShotKeyState OneShot::state_[OneShot::ONESHOT_KEY_COUNT];
+uint16_t OneShot::start_time_ = 0;
+KeyAddr OneShot::prev_key_addr_ = UnknownKeyswitchLocation;
```

Comment:
Instead of keeping track of the `Key` value of the last key pressed, I changed it to the address of the key, which is more certainly unique, and needs only one byte of RAM. This does mean that two quick taps (or even rollover) of a pair of identically-mapped OneShot keys will no longer result in that OneShot modifier/layer becoming sticky. This is worth considering, even though it's not mentioned in the documentation of the behaviour, because there might be users out there who make an OSL sticky by, for example, simultaneously tapping both palm keys on a Model01. That could become officially supported as a second way of making a OneShot sticky, but I have not done so yet.